python/dgl/dataloading/dataloader.py

```diff
@@ -46,6 +46,81 @@ def _locate_eids_to_exclude(frontier_parent_eids, exclude_eids):
         result = np.isin(frontier_parent_eids, exclude_eids).nonzero()[0]
         return F.zerocopy_from_numpy(result)
 
+class _EidExcluder():
+    def __init__(self, exclude_eids):
+        device = None
+        if isinstance(exclude_eids, Mapping):
+            for _, v in exclude_eids.items():
+                if device is None:
+                    device = F.context(v)
+                    break
+        else:
+            device = F.context(exclude_eids)
+        self._exclude_eids = None
+        self._filter = None
+
+        if device == F.cpu():
```

Comment:
Maybe add a TODO for CPU?  If the filters are implemented for CPU then we won't use `_locate_eids_to_exclude` right?

Reply:
Added in 
aa21891.